src/write.jl

```diff
@@ -0,0 +1,621 @@
+struct JSONWriteStyle <: JSONStyle end
+
+sizeguess(::Nothing) = 4
+sizeguess(x::Bool) = 5
+sizeguess(x::Integer) = 20
+sizeguess(x::AbstractFloat) = 20
+sizeguess(x::Base.IEEEFloat) = Base.Ryu.neededdigits(typeof(x))
```

Comment:
From what I can tell, both `Base.IEEEFloat` and `Base.Ryu.neededdigits` are not public.

Reply:
I've changed `IEEEFloat` to the Union as is defined in Base. We'll have to open another discussion on the `Ryu` usage; it would be great to more formally expose some of that functionality as part of the public API of Base.